bin/kano_tzupdate

```diff
@@ -0,0 +1,156 @@
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE 
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER 
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, 
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE 
+# SOFTWARE. 
+#
+#
+#
+# Copyright (C) 2015 Kano Computing Ltd.
+# License: http://www.gnu.org/licenses/gpl-2.0.txt GNU General Public License v2
+#
+# This script has been adapted on KanoOS to provide for custom authentication
+# against geonames, and getting current gelocation from the free webservice ipinfo.io
+#
+
+import argparse
+import errno
+import json
+import os
+import sys
+
+
+# Accomodate a library to speak to the web services
+try:
+    from urllib.request import urlopen
+    from urllib.parse import urlencode
+except ImportError:  # Python 2 fallback
+    from urllib import urlopen, urlencode
+
+
+# Authentication information required by the web services
+AUTH = {
+    "geonames": {
+        "username": "kanosystem",
+    },
+}
+
+PATH = {
+    "zoneinfo": "/usr/share/zoneinfo",
+    "localtime": "/etc/localtime",
+}
+
+
+def get_geolocation(ip=None):
+    '''
+    This is a simplified version to obtain your current location
+    by using the free web service ipinfo.io API
+    '''
+    lat=lng=None
+
+    if ip:
+        url='http://ipinfo.io/{}/geo'.format(ip)
+    else:
+        url="http://ipinfo.io/geo"
+
+    try:
+        result = urlopen(url)
+        res_data = json.loads(result.read())
+        lat,lng=res_data['loc'].split(',')
+        return lat,lng
+    except:
+        raise
+
+
+def timezone_from_location(coords):
+    """
+    Return the timezone for a set of coordinates using the GeoNames API.
+
+    :param coords: latitude and longitude
+    :returns: timezone, or None if none is available
+    """
+
+    url = "http://api.geonames.org/timezoneJSON?"
+
+    latitude, longitude = coords
+    url_params = {
+        "username": AUTH["geonames"]["username"],
+        "lat": latitude,
+        "lng": longitude,
+    }
+
+    res_handle = urlopen(url + urlencode(url_params))
+
+    try:
+        res_encoding = res_handle.headers.get_content_charset()
+    except AttributeError:  # Python 2 fallback
+        res_encoding = res_handle.headers.getparam("charset")
+
+    res_data = json.loads(res_handle.read().decode(res_encoding))
+
+    timezone = res_data.get("timezoneId")
+
+    return timezone
+
+
+if __name__ == "__main__":
+    parser = argparse.ArgumentParser(description=__doc__)
+    parser.add_argument(
+        "-p", "--print-only",
+        action="store_true",
+        help="print the timezone, but don't update " + PATH["localtime"]
+    )
+    parser.add_argument(
+        "-a", "--ip",
+        help="use this IP instead of automatically detecting it"
+    )
+    args = parser.parse_args()
+
+    try:
+        coords = get_gelocation(ip=args.ip)
```

Comment:
Okay this looks good to merge to me!


On 17 March 2015 at 10:02, Albert Casals <notifications@github.com> wrote:

> In bin/kano_tzupdate
> <https://github.com/KanoComputing/kano-toolset/pull/116#discussion_r26560568>
> :
>
> > +
> > +if __name__ == "__main__":
> > +    parser = argparse.ArgumentParser(description=__doc__)
> > +    parser.add_argument(
> > +        "-p", "--print-only",
> > +        action="store_true",
> > +        help="print the timezone, but don't update " + PATH["localtime"]
> > +    )
> > +    parser.add_argument(
> > +        "-a", "--ip",
> > +        help="use this IP instead of automatically detecting it"
> > +    )
> > +    args = parser.parse_args()
> > +
> > +    try:
> > +        coords = get_gelocation(ip=args.ip)
>
> that's right @Ealdwulf <https://github.com/Ealdwulf> thanks!
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/KanoComputing/kano-toolset/pull/116/files#r26560568>.
>